### PR TITLE
TWI-16 Create tweet

### PR DESCRIPTION
**Pull request by pm888**

### CODE REVIEW

File: internal/tweets/tweets.go [hunk 1]
```diff
@@ -13,47 +14,48 @@ import (
...
 	cookie, err := r.Cookie("session")
-	if err != nil || cookie.Value == "" {
-		http.Error(w, "Unauthorized", http.StatusUnauthorized)
+	var sessionID string
+	if apikey != "" {
+		sessionID = apikey
+	} else {
+		sessionID = cookie.Value
+	}
+	if apikey == "" && (err != nil || cookie == nil) {
```
Comment (bunin) [inline internal/tweets/tweets.go]: why is this here?

**Review by bunin — CHANGES_REQUESTED** [review]

File: internal/tweets/creatTweet.go
Comment (bunin) [inline internal/tweets/creatTweet.go:1]: there is a typo in a filename

File: internal/tweets/creatTweet.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	"net/http"
+	"time"
+)
+
+func CreateNewTweet(w http.ResponseWriter, r *http.Request) {
+	tweetValid := &TweetValid{
+		Validate: validator.New(),
+		ValidErr: make(map[string]string),
+	}
+	if err := RegisterUsersValidations(tweetValid); err != nil {
```
Comment (bunin) [inline internal/tweets/creatTweet.go]: does it really have to do anything with "Users"?

File: internal/tweets/tweetsStr.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	Public              bool      `json:"public"`
+	OnlyFollowers       bool      `json:"only_followers"`
+	OnlyMutualFollowers bool      `json:"only_mutual_followers"`
+	OnlyMe              bool      `json:"only_me"`
+	LoginToken          string
+	ParentTweetId       int `json:"parent_tweet_id"`
+}
+type CreatNewTweet struct {
+	TweetID             int
+	Text                string `json:"text" validate:"omitempty,checkTweetText"`
```
Comment (bunin) [inline internal/tweets/tweetsStr.go]: that would mean that empty text is ok, which I think is not

**Review by bunin — CHANGES_REQUESTED** [review]

File: internal/tweets/tweets.go (hunk 1 above)
Comment (pm888) [reply internal/tweets/tweets.go]: done

File: internal/tweets/creatTweet.go
Comment (pm888) [reply internal/tweets/creatTweet.go:1]: done

File: internal/tweets/creatTweet.go (hunk 2 above)
Comment (pm888) [reply internal/tweets/creatTweet.go]: done

File: internal/tweets/tweetsStr.go (hunk 3 above)
Comment (pm888) [reply internal/tweets/tweetsStr.go]: done

File: internal/tweets/createTweet.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	"time"
+)
+
+func CreateNewTweet(w http.ResponseWriter, r *http.Request) {
+	tweetValid := &TweetValid{
+		Validate: validator.New(),
+		ValidErr: make(map[string]string),
+	}
+	if err := RegisterTweetValidations(tweetValid); err != nil {
+		services.ReturnErr(w, err.Error(), http.StatusBadRequest)
```
Comment (bunin) [inline internal/tweets/createTweet.go]: it has nothing to do with a request

File: openapi.yaml [hunk 5]
```diff
@@ -412,26 +416,52 @@ components:
...
       in: header
       name: X-API-KEY
-    appId:
-      type: apiKey
-      in: header
-      name: X-APP-ID
 security:
     - apiKey: [ ]
-      appId: [ ]
+      appId: [ ]
```
Comment (bunin) [inline openapi.yaml:467]: what's appId and why is it here?

**Review by bunin — APPROVED** [review]

File: openapi.yaml (hunk 5 above)
Comment (pm888) [reply openapi.yaml:467]: authorize

Comment (bunin) [reply openapi.yaml:467]: authorize what?
